app/src/lib/git/cherry-pick.ts

```diff
@@ -80,6 +184,111 @@ function parseCherryPickResult(result: IGitResult): CherryPickResult {
   }
 }
 
+/**
+ * Inspect the `.git/sequencer` folder and convert the current cherry pick
+ * state into am `ICherryPickProgress` instance as well as return an array of
+ * remaining commits queued for cherry picking.
+ *  - Progress instance required to display progress to user.
+ *  - Commits required to track progress after a conflict has been resolved.
+ *
+ * This is required when Desktop is not responsible for initiating the cherry
+ * pick and when continuing a cherry pick after conflicts are resolved:
+ *
+ * It returns null if it cannot parse an ongoing cherry pick. This happens when,
+ *  - There isn't a cherry pick in progress (expected null outcome).
+ *  - Runs into errors parsing cherry pick files. This is expected if cherry
+ *    pick is aborted or finished during parsing. It could also occur if cherry
+ *    pick sequencer files are corrupted.
+ */
+export async function getCherryPickSnapShot(
+  repository: Repository
+): Promise<ICherryPickSnapshot | null> {
+  const cherryPickHead = readCherryPickHead(repository)
+  if (cherryPickHead === null) {
+    // If there no cherry pick head, there is no cherry pick in progress.
+    return null
+  }
+
+  let firstSha: string = ''
+  let lastSha: string = ''
+  const remainingShas: string[] = []
+  // Try block included as files may throw an error if it cannot locate
+  // the sequencer files. This is possible if cherry pick is continued
+  // or aborted at the same time.
+  try {
+    // This contains the sha of the first committed pick.
+    firstSha = (
+      await FSE.readFile(
+        Path.join(repository.path, '.git', 'sequencer', 'abort-safety'),
+        'utf8'
+      )
+    ).trim()
+
+    if (firstSha === '') {
+      // Technically possible if someone continued or aborted the cherry pick at
+      // the same time
+      return null
+    }
+
+    // This contains a reference to the remaining commits to cherry pick.
+    // Each line is of the format: `pick shortSha commitSummary`
+    const remainingPicks = (
+      await FSE.readFile(
+        Path.join(repository.path, '.git', 'sequencer', 'todo'),
+        'utf8'
+      )
+    ).trim()
+
+    if (remainingPicks === '') {
+      // Technically possible if someone continued or aborted the cherry pick at
+      // the same time
+      return null
+    }
+
+    remainingPicks.split('\n').forEach(line => {
+      const linePieces = line.split(' ')
+      if (linePieces.length > 2) {
+        remainingShas.push(linePieces[1])
+      }
+    })
```

Comment:
It might be worth adding a comment with the expected format of the contents of this `todo` file to understand why it's parsing like it is 🤔

Reply:
🙂  I do have on line 234, but I will move it down closer to here since this is the part really parsing it.

Reply:
Ouch! Sorry, I missed that 😅